R/vsc.R

```diff
@@ -7,6 +7,64 @@ homedir <- Sys.getenv(
 dir_watcher <- Sys.getenv("VSCODE_WATCHER_DIR", file.path(homedir, ".vscode-R"))
 request_file <- file.path(dir_watcher, "request.log")
 request_lock_file <- file.path(dir_watcher, "request.lock")
+settings_file <- file.path(dir_watcher, "settings.json")
+user_options <- names(options())
+
+load_settings <- function() {
+  if (!file.exists(settings_file)) {
+    return(FALSE)
+  }
+
+  mapping <- c(
+    "plot.useHttpgd" = "vsc.use_httpgd",
+    "workspaceViewer.showObjectSize" = "vsc.show_object_size",
+    "session.emulateRStudioAPI" = "vsc.rstudioapi",
+    "session.levelOfObjectDetail" = "vsc.str.max.level",
+    "session.watchGlobalEnvironment" = "vsc.globalenv",
+    "session.viewers.viewColumn.plot" = "vsc.plot",
+    "session.viewers.viewColumn.browser" = "vsc.browser",
+    "session.viewers.viewColumn.viewer" = "vsc.viewer",
+    "session.viewers.viewColumn.pageViewer" = "vsc.page_viewer",
+    "session.viewers.viewColumn.view" = "vsc.view",
+    "session.viewers.viewColumn.helpPanel" = "vsc.helpPanel"
+  )
+
+  vsc_settings <- tryCatch(jsonlite::read_json(settings_file), error = function(e) {
+    message("Error occurs when reading VS Code settings: ", conditionMessage(e))
+  })
+
+  if (is.null(vsc_settings)) {
+    return(FALSE)
+  }
+
+  vsc_settings <- unlist(vsc_settings)
```

Comment:
It is already a nested list. What we need of `unlist` here is to convert `list(a=list(b=list(c="x")))` to `c(a.b.c = "x")`  so that it is easier to do the mapping below.

Reply:
If we need a list-valued item to work in R at some point, then I guess we need to avoid `unlist` here.

Reply:
Ah ok, thanks for the clarification

Reply:
I rewrite the `mapping` using `alist` with expressions so that `unlist` is no longer needed and list value could be supported too.